ement-room.el

```diff
@@ -2419,6 +2419,59 @@ these all require at least version 29 of Emacs):
                                ('remove "removed from"))
                              (ement--format-room space))))))
 
+;;;;; Admin/power commands
+
+(defun ement-room--kick-ban-interactive (prompt)
+  "Interactive arguments for kicking or un/banning a user.
+See `ement-room-kick-user', `ement-room-ban-user', `ement-room-unban-user'.
+Returns (user room session reason)."
+  (ement-room-with-highlighted-event-at (point)
+    (let* ((room ement-room)
+           (event (ewoc-data (ewoc-locate ement-ewoc)))
+           (user (ement-event-sender event)))
+      (if (yes-or-no-p (format "%s user %s <%s> from room %s? "
+                               prompt
+                               (ement-user-displayname user)
+                               (ement-user-id user)
+                               (ement--format-room room)))
+          (list user ement-room ement-session
+                (read-string "Reason (optional): "
+                             nil nil nil 'inherit-input-method))
+        ;; HACK: This isn't really an error, but is there a cleaner way to cancel?
+        (user-error "Aborted")))))
+
+(defun ement-room--kick-ban-user (type user room session reason successfmt)
```

Comment:
I think it might be better to pass callback functions (i.e. `(&key then else)`) so we could potentially do other things after the request completes or fails.

Also, maybe the reason should be an optional argument.  What do you think?

Reply:
I think it's ok for `reason` to be required here, as it's an internal function.

I don't have firm thoughts on using callbacks here.  I don't think it benefits the current functionality, so it could possibly be a refactoring if a use-case arose?

The only thing which springs to mind is your earlier suggestion of "a convenience feature to also remove all recent events by the banned user" which could be a `:then` callback for the `ban` case.

So perhaps it can take optional `:then` and `:else` callbacks, and use the current function as `:then` if it wasn't supplied?

Reply:
You're right that it doesn't benefit us at the moment, but I think it would be more consistent with the design of similar commands; and as you noted, it would enable further convenience features.  I agree that the callbacks could be optional, i.e. they could have as default values functions which just print the messages as it currently does.

Reply:
Done.